tools/run_tests/artifact_targets.py

```diff
@@ -99,8 +103,47 @@ def build_jobspec(self):
     if self.platform == 'linux':
       if self.arch == 'x86':
         environ['SETARCH_CMD'] = 'linux32'
+      # Inside the manylinux container, the python installations are located in
+      # special places...
+      environ['PYTHON'] = '/opt/python/{}/bin/python'.format(self.manylinux_build)
+      environ['PIP'] = '/opt/python/{}/bin/pip'.format(self.manylinux_build)
+      # Our docker image has all the prerequisites pip-installed already.
+      environ['SKIP_PIP_INSTALL'] = '1'
+      # Platform autodetection for the manylinux1 image breaks so we set the
+      # defines ourselves.
+      # TODO(atash) get better platform-detection support in core so we don't
+      # need to do this manually...
+      environ['CFLAGS'] = " ".join([
```

Comment:
Well, to be fair, this is the purpose of `GPR_NO_AUTODETECT_PLATFORM`, so that isn't so ugly if that was a third party package that we don't want to support. BUT, for the sake of maintainability, this needs indeed to be in port_platform.h. I recommend adding `-DGPR_MANYLINUX` here, and moving everything into an `#ifdef GPR_MANYLINUX` on the top of port_platform.h.